Optimizer dev

sergey-tomin opened the pull request:
- added DeviceManager class as an intermediary between devices and Optimizer. it is needed to be able to run optimization with virtual devices. 
- added logger to some classes. need to catch some strange bug. 
- added petra and flash machine interfaces. 
- added changes to improve CLI to run a sequence of optimizations. 
- added example with a sequence of optimization

sergey-tomin committed:
added DeviceManager to mint.py. The object is an intermediary between Devices and Optimizer.

sergey-tomin committed:
merged with BKR version

sergey-tomin committed:
minor

sergey-tomin committed:
fixed bug with logbook

sergey-tomin committed:
changes to break link between Optimizer and Action

sergey-tomin committed:
Merge pull request #1 from ocelot-collab/optimizer-dev

Optimizer dev

sergey-tomin committed:
added FLASH machine interface

sergey-tomin committed:
added logging to some files.

sergey-tomin committed:
added example of sequence of optimizations

hhslepicka reviewed (changes requested):
Thank you for the changes Sergey.
I am having a bit of problem understanding why do you need the device manager.
I get the idea of it I am just not sure how you will use it to accomplish what was written at the class docstring.
Let’s schedule a call for some time next week and go over the changes.

sergey-tomin committed:
fixed bug when it was not possible to stop optimization if alarm channel was out of limits.

sergey-tomin committed:
minor changes

sergey-tomin committed:
fixed bug when GUI calls get_value(ch) even if ch = ""

sergey-tomin committed:
added logging to xfel_interface

sergey-tomin committed:
small optimization of the resetpanel. do not update spinboxes if previous limit condition is the same as current.

sergey-tomin committed:
added description of a method in Device

sergey-tomin committed:
Minor improvement - a reduced time of reaction of the Optimizer class on the stop signal. The delay appears in case the averaging of the target function is used (multiple readings of the target function in a loop).

sergey-tomin committed:
fixed inconsistency in font size in different functions.

sergey-tomin committed:
Merge pull request #2 from ocelot-collab/optimizer-dev

Optimizer dev

sergey-tomin committed:
linac4 interface

sergey-tomin committed:
Merge branch 'optimizer-dev' of https://github.com/sergey-tomin/optimizer into optimizer-dev

sergey-tomin committed:
added check block to prevent Optimizer crush if someone wants to play with browser before the optimizations finishes its work.

sergey-tomin committed:
minor

sergey-tomin committed:
fixed bug with Alarm device

sergey-tomin committed:
renamed DeviceManager to MetaDevice. It is better reflect idea of the class

sergey-tomin committed:
refactoring MachineStatus functionality

sergey-tomin committed:
Merge pull request #3 from ocelot-collab/optimizer-dev

Optimizer dev

hhslepicka commented:
TODO before Merging:
- [x] Add mention at docstrings for `MetaDevice` stating that it is only to be used via scripted optimization and it is not valid for UI optimizations.
- [x] Rename at `mint.py` at line 246 ->  `self.dev_manager = MetaDevice()` to match the new class name `MetaDevice`.

sergey-tomin committed:
Added mention at docstrings that MetaDevice is only to be used via scripted optimization and it is not valid for UI optimizations.
  Renamed at mint.py at self.meta_dev = MetaDevice() to match the new class name MetaDevice.

sergey-tomin commented:
Thanks for remarks. Fixed and ready for merge. Please have a look.
